src/hooks/wallets/useOnboard.ts

```diff
@@ -124,16 +124,16 @@ export const useInitOnboard = () => {
 
   // Disable unsupported wallets on the current chain
   useEffect(() => {
-    if (!onboard || !chain?.disabledWallets) return
+    if (!onboard || !chain) return
 
     const enableWallets = async () => {
       const { getSupportedWallets } = await import('@/hooks/wallets/wallets')
-      const supportedWallets = getSupportedWallets(chain.disabledWallets)
+      const supportedWallets = getSupportedWallets(chain)
       onboard.state.actions.setWalletModules(supportedWallets)
     }
 
     enableWallets()
-  }, [chain?.disabledWallets, onboard])
+  }, [chain, onboard])
```

Comment:
The previous logic is better as it narrows more the dependency array, no?

Reply:
We are now not just relying on the `disabledWallets` but also the `rpcUri` within `getSupportedWallets`. The `chain` only changes when switching chain anyway. It should act the same